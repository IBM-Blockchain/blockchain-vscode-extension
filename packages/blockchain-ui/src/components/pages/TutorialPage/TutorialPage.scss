@import '../../../mixins/layout-mixins.scss';
@import '@carbon/colors/scss/colors';

.tutorial-page-container {
    @include default-page-styling;
    margin-left: $spacing-05;
    margin-top: $spacing-05;
}
