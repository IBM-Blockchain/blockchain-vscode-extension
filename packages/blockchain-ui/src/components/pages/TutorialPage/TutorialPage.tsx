import React, { Component } from 'react';
import HeadingCombo from '../../elements/HeadingCombo/HeadingCombo';
import './TutorialPage.scss';

interface IProps {
    tutorialData: Array<{seriesName: string, seriesTutorials: any[]}>;
}

class TutorialPage extends Component<IProps> {
    render(): JSX.Element {
        return (
            <div className='bx--grid tutorial-page-container'>
                <div className='bx--row'>
                    <HeadingCombo
                        headingText='Blockchain Tutorials'
                        subheadingText='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
                    />
                </div>
                <div className='bx--row'>
                    {/* this is where the tutorials will go */}
                </div>
            </div>
        );
    }
}

export default TutorialPage;
