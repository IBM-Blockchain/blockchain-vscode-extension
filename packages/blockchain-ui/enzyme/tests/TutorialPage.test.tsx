import React from 'react';
import renderer from 'react-test-renderer';
import chai from 'chai';
import sinonChai from 'sinon-chai';

import TutorialPage from '../../src/components/pages/TutorialPage/TutorialPage';

chai.should();
chai.use(sinonChai);

describe('TutorialPage component', () => {

    const tutorialData: Array<{seriesName: string, seriesTutorials: any[]}> = [
        {
            seriesName: 'Basic tutorials',
            seriesTutorials: [
                {
                    title: 'a1',
                    length: '4 weeks',
                    file: 'some/file/path'
                }
            ]
        },
        {
            seriesName: 'Other tutorials',
            seriesTutorials: [
                {
                    title: 'something really interesting',
                    length: '10 minutes',
                    file: 'another/file/path'
                }
            ]
        }
    ];

    it('should render the expected snapshot', () => {
        const component: any = renderer
            .create(<TutorialPage tutorialData={tutorialData}/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
