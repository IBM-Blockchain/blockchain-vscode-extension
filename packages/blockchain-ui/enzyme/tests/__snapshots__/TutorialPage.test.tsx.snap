// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TutorialPage component should render the expected snapshot 1`] = `
<div
  className="bx--grid tutorial-page-container"
>
  <div
    className="bx--row"
  >
    <div
      className="heading-combo-container"
    >
      <h3>
        Blockchain Tutorials
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </p>
    </div>
  </div>
  <div
    className="bx--row"
  />
</div>
`;
